src/plugins/intel_cpu/src/cpu_memory.h

```diff
@@ -89,18 +89,24 @@ class MemoryMngrWithReuse : public IMemoryMngr {
     static void destroy(void *ptr);
 };
 
+class IMemoryMngrObserver : public IMemoryMngr {
+public:
+    virtual void registerMemory(Memory* memPtr) = 0;
+    virtual void unregisterMemory(Memory* memPtr) = 0;
```

Comment:
@maxnick  Why do we need the registerMemory/unregisterMemory inferface in IMemoryMngrObserver? Sounds they are for DnnlMemoryMngr only.  I don't see where they are actually used in PartitionedMemoryMngr.

Reply:
`PartitionedMemoryMngr` dispatches these calls to the underlining  `DnnlMemoryMngr `, that is how the decorator pattern works.